wemake_python_styleguide/checker.py

```diff
@@ -56,6 +69,8 @@ def run(self) -> Generator[CheckResult, None, None]:
         """
         for visitor_class in self._visitors:
             visiter = visitor_class()
+            if hasattr(visiter, 'provide_options'):
+                visiter.provide_options(self.options)
```

Comment:
The only thing that bothers me here is: why this API is not required?

We can define a basic `provide_options` call in `BaseVisiter` class.
And just call it.